src/main/java/com/ms/zookeeper/authz/ZkUserAuthorizer.java

```diff
@@ -0,0 +1,75 @@
+package com.ms.zookeeper.authz;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+
+public class ZkUserAuthorizer implements ZkAuthorizer {
+
+    private static final Logger LOG =
+            Logger.getLogger(ZkUserAuthorizer.class);
+
+    private static  ZkUserAuthorizer instance = null;
+
+    private static final String HOST_PREFIX = "host/";
+
+    private static final String ARBITRARY_HOST = HOST_PREFIX + "*";
+
+    private final Set<String> validRealms;
+
+    public ZkUserAuthorizer() {
+        String realmsString = System.getProperty("zookeeper.validKerberosRealms", "");
+        validRealms = new HashSet<String>(Arrays.asList(realmsString.split(",")));
+        LOG.info("Valid Kerberos Realms: " + validRealms.toString());
+    }
+
+    public static ZkUserAuthorizer getInstance() {
+        if (instance == null) {
+            instance = new ZkUserAuthorizer();
+        }
+        return instance;
+    }
+
+    private boolean isValidRealm(final String realm) {
```

Comment:
why do you care about valid realm? Can't the file just contain full principal, including the realm?